dietpi/dietpi-software

```diff
@@ -10811,6 +10878,76 @@ location = /.well-known/caldav  { return 301 /baikal/html/dav.php; }' > /etc/ngi
 
 		fi
 
+		software_id=125 # Synapse Server
+		if (( ${aSOFTWARE_INSTALL_STATE[$software_id]} == 1 )); then
+
+			Banner_Configuration
+
+			G_EXEC mkdir -p /mnt/dietpi_userdata/synapse
+
+			local synapse_pass=$(tr -dc '[:alnum:]' < /dev/random | head -c30)
+
+			G_EXEC sudo -u postgres psql -c "create role synapse with login password '$synapse_pass';"
+			G_EXEC sudo -u postgres createdb --encoding=UTF8 --locale=C --template=template0 --owner=synapse synapse
```

Comment:
Will this fail when the database exists already? In case we should skip it then.

Reply:
It fails.
The following can be used to check whether a role exists already:
```sh
sudo -u postgres psql -c '\du' | grep -q '^[[:blank:]]*synapse[[:blank:]]'
```
And the following could be used to check whether the database exists already:
```sh
sudo -u postgres psql -l -t | grep -q '^[[:blank:]]*synapse[[:blank:]]'
```

If the role exists already, I think we should change it's password instead and, if the config does already exist, re-apply it. It's a little nasty sed required to apply it to the correct place. I can push that commit.

The servername and analytics prompts, as well as register_new_matrix_user should be as well only called when the config file does not yet exist. The latter else fails.